src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -1727,33 +1727,49 @@ public LazyFixedStatement(Lazy<IVariableDeclarationStatement> variables, Lazy<IO
     }
 
     /// <summary>
-    /// Represents a C# foreach statement or a VB For Each statement.
+    /// Represents a C# 'foreach' statement or a VB 'For Each' staement.
     /// </summary>
     internal abstract partial class BaseForEachLoopStatement : LoopStatement, IForEachLoopStatement
     {
-        public BaseForEachLoopStatement(ILocalSymbol iterationVariable, LoopKind loopKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-            base(loopKind, OperationKind.LoopStatement, semanticModel, syntax, type, constantValue, isImplicit)
+        public BaseForEachLoopStatement(ILocalSymbol iterationVariable, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
+            base(LoopKind.ForEach, OperationKind.LoopStatement, semanticModel, syntax, type, constantValue, isImplicit)
         {
             IterationVariable = iterationVariable;
         }
         /// <summary>
         /// Iteration variable of the loop.
         /// </summary>
         public ILocalSymbol IterationVariable { get; }
+        protected abstract IOperation LoopControlVariableImpl { get; }
         protected abstract IOperation CollectionImpl { get; }
+        protected abstract ImmutableArray<IOperation> AtLoopBottomExpressionListImpl { get; }
         public override IEnumerable<IOperation> Children
         {
             get
             {
+                yield return LoopControlVariable;
```

Comment:
This raises an interesting point to me: in IOperation, we haven't stated that a child node can potentially be null. Do we want to be returning null elements in the Children property? I don't think we should.

Reply:
With the new proposed API from @AlekseyTs, loop control variable should not be null, except for error cases. I agree that we should filter it out when it is null.

Reply:
I'd be willing to bet that we have other cases where we'll be returning null elements in children, so I filed https://github.com/dotnet/roslyn/issues/21849 to sanity check all of our APIs for this.